### PR TITLE
[SUP-703]: Handle relative url for icon and image url.

### PR DESCRIPTION
**Pull request by GuelorEmanuel**
Added a case for relative based URL e.g `//yourdomain.com/images/example.png`

### CODE REVIEW

File: previews/model.py [hunk 1]
```diff
@@ -124,6 +124,8 @@ def fetch(self, timeout=1):
 
                 if icon_href.find("http") != -1:
                     self.icon = icon_href
+                elif icon_href.startswith("//"):
+                    self.icon = "http:" + icon_href
                 else:
                     self.icon = self.url + icon_href
```
Comment (wole-ada) [inline previews/model.py:130]: Hi @GuelorEmanuel 

I think there's one more issue here.
For example, some sites could have the favicon URL as a relative path eg `/assets/favicon.ico`.
This URL is relative to the base URL eg `https://www.mysite.com` (so the favicon path should be
https://www.mysite.com/assets/favicon.ico)

Assuming the `self.url` here is something like `https://www.mysite.com/some/path`, then the favicon url
will be `https://www.mysite.com/some/path/assets/favicon.ico` which will return a 404.
I think the `self.url` here should be the base URL without any additional paths.

Comment (wole-ada) [reply previews/model.py:130]: `self.icon = self.base_url + icon_href` if the `icon_href` is a relative path
would need to figure out how to extract the base url from `self.url`

Comment (wole-ada) [reply previews/model.py:130]: Real world example:
URL = `https://www.sydneycreditunion.com/Home/ProductsAndServices/YourFuture/Investing/`
Icon URL = `/DynamicContent/Resources/Images/favicon.ico`
Full Icon URL = `https://www.sydneycreditunion.com/Home/ProductsAndServices/YourFuture/Investing/DynamicContent/Resources/Images/favicon.ico`
Base URL + Icon URL = `https://www.sydneycreditunion.com/DynamicContent/Resources/Images/favicon.ico`

Comment (GuelorEmanuel) [reply previews/model.py:130]: I see, I was just thinking of handling the case when the icon URL looks something like this`//cdn.shopify.com/s/files/1/0678/8333/t/335/assets/favicons-optimized.png` and didn't catch that issue at all. Thank you for pointing that out and for the thorough explanation :pray:

**Review by wole-ada — APPROVED** [review]

**Pull request merged**